core/process.py

```
@@ -92,14 +105,24 @@ def start_pipecat_process(
-        command,
-        env=os.environ.copy(),  # Copy the current environment
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        text=True,  # Capture output as text
-    )
+    child_env = os.environ.copy()
+    if mcp_runtime_headers:
+        child_env["MCP_RUNTIME_HEADERS_JSON"] = json.dumps(mcp_runtime_headers)
+
+    try:
+        process = subprocess.Popen(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)